File: monitorlibrary/lib/data/geofence_event.dart
import 'package:meta/meta.dart';
import 'package:monitorlibrary/data/user.dart';

class GeofenceEvent {
  String? status, geofenceEventId, date, projectPositionId, projectName, userId;
  User? user;
  GeofenceEvent(
      {required this.status,
        required this.user,
        required this.geofenceEventId,
        required this.projectPositionId,
        required this.projectName, this.userId,
        required this.date});

  GeofenceEvent.fromJson(Map data) {
    this.status = data['status'];
    this.userId = data['userId'];
    this.geofenceEventId = data['geofenceEventId'];
    this.projectPositionId = data['projectPositionId'];
    this.projectName = data['projectName'];
    this.date = data['date'];
    if (data['user'] != null) {
      this.user = User.fromJson(data['user']);
    } 
  }
  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {
      'status': status,
      'userId': userId,
      'geofenceEventId': geofenceEventId,
      'projectPositionId': projectPositionId,
      'projectName': projectName,
      'date': date,
      'user': user == null? null : user!.toJson(),
    };
    return map;
  }
}

File: backend/src/main/java/com/monitor/backend/data/GeofenceEvent.java
package com.monitor.backend.data;

import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "geofenceEvent")
public class GeofenceEvent {

       private String status, geofenceEventId, date, projectPositionId, projectName, userId;
       private User user;

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getGeofenceEventId() {
        return geofenceEventId;
    }

    public void setGeofenceEventId(String geofenceEventId) {
        this.geofenceEventId = geofenceEventId;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getProjectPositionId() {
        return projectPositionId;
    }

    public void setProjectPositionId(String projectPositionId) {
        this.projectPositionId = projectPositionId;
    }

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}
